docs/csharp/programming-guide/generics/constraints-on-type-parameters.md

```diff
@@ -18,22 +18,31 @@ Constraints inform the compiler about the capabilities a type argument must have
+
+Some constraints are mutually exclusive, and some constraints must be in a specified order:
+
+- You can apply at most one of the `struct`, `class`, `class?`, `notnull`, `unmanaged`. If you supply any of these constraints, it must be the first constraint specified for that type parameter.
+- That base class constraint, (`where T : Base` or `where T : Base?`), can't be combined with any of `struct`, `class`, `class?`, `notnull`, or `unmanaged`.
+- You can apply at most one base class constraint, in either form. If you want to support the nullable base type, use `Base?`.
+- You can't name both the non-nullable and nullable form of an interface as a constraint.
+- The `new()` constraint can't be combined with the `struct` or `unmanaged` constraints. If you specify the `new()` constraint, it must be the last constraint for that type parameter.
+- The `default` constraint can be applied only on override or explicit interface implementations. It can't be combined with either the `struct` or `class` constraints.
 
 ## Why use constraints
 
-Constraints specify the capabilities and expectations of a type parameter. Declaring those constraints means you can use the operations and method calls of the constraining type. If your generic class or method uses any operation on the generic members beyond simple assignment or calling any methods not supported by <xref:System.Object?displayProperty=nameWithType>, you'll apply constraints to the type parameter. For example, the base class constraint tells the compiler that only objects of this type or derived from this type will be used as type arguments. Once the compiler has this guarantee, it can allow methods of that type to be called in the generic class. The following code example demonstrates the functionality you can add to the `GenericList<T>` class (in [Introduction to Generics](../../../standard/generics/index.md)) by applying a base class constraint.
+Constraints specify the capabilities and expectations of a type parameter. Declaring those constraints means you can use the operations and method calls of the constraining type. You apply constraints to the type parameter when your generic class or method uses any operation on the generic members beyond simple assignment including calling any methods not supported by <xref:System.Object?displayProperty=nameWithType>. For example, the base class constraint tells the compiler that only objects of this type or derived from this type can replace that type argument. Once the compiler has this guarantee, it can allow methods of that type to be called in the generic class. The following code example demonstrates the functionality you can add to the `GenericList<T>` class (in [Introduction to Generics](../../../standard/generics/index.md)) by applying a base class constraint.
```

Comment:
```suggestion
Constraints specify the capabilities and expectations of a type parameter. Declaring those constraints means you can use the operations and method calls of the constraining type. You apply constraints to the type parameter when your generic class or method uses any operation on the generic members beyond simple assignment, which includes calling any methods not supported by <xref:System.Object?displayProperty=nameWithType>. For example, the base class constraint tells the compiler that only objects of this type or derived from this type can replace that type argument. Once the compiler has this guarantee, it can allow methods of that type to be called in the generic class. The following code example demonstrates the functionality you can add to the `GenericList<T>` class (in [Introduction to Generics](../../../standard/generics/index.md)) by applying a base class constraint.
```